provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/AbstractRuleViolationNotificationMessageExtractor.java

```diff
@@ -75,13 +78,27 @@ private BomComponentDetails createBomComponentDetails(BlackDuckApiClient blackDu
             bomComponent404Handler.logIf404OrThrow(e, componentVersionStatus.getComponentName(), componentVersionStatus.getComponentVersionName());
             return bomComponentDetailsCreator.createMissingBomComponentDetails(
                 componentVersionStatus.getComponentName(),
-                componentVersionStatus.getComponent(),
+                createComponentUrl(componentVersionStatus),
                 componentVersionStatus.getComponentVersionName(),
-                componentVersionStatus.getComponentVersion(),
+                createComponentVersionUrl(componentVersionStatus),
```

Comment:
Fixes and issue where we would fall-back to the KB Component link instead of first attempting to construct the link from the BOM Component URL. This was causing issue-tracker searches to miss existing issues.